packages/runtime-common/realm.ts

```diff
@@ -1710,6 +1739,19 @@ export class Realm {
   get isPublicReadable(): boolean {
     return this.#permissions['*']?.includes('read') ?? false;
   }
+
+  publicEndpoints = [
+    { path: '/_session', method: 'POST' },
+    { path: '/_readiness-check', method: 'GET' },
+  ];
+
+  private isRequestToPublicEndpoint(request: Request) {
+    return !!this.publicEndpoints.find(
+      (endpoint) =>
+        request.url.endsWith(endpoint.path) &&
```

Comment:
Is this vulnerable to an attack where a query string or anchor tag ends with "/_session" and thus is treated as a public endpoint by this method?

Reply:
Good catch—we can use `new URL(request.url).path.startsWith(endpoint.path)` instead

Reply:
I refactored this to to use `lookupRouteTable` from `router.ts` instead, in https://github.com/cardstack/boxel/pull/1201/commits/09e231b501e147b466bfe4d5086af205bd0610f1. Should make route checking more thorough